specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json

```diff
@@ -11249,6 +11277,11 @@
           "title": "The command line of the Job Preparation task.",
           "description": "The command line does not run under a shell, and therefore cannot take advantage of shell features such as environment variable expansion. If you want to take advantage of such features, you should invoke the shell in the command line, for example using \"cmd /c MyCommand\" in Windows or \"/bin/sh -c MyCommand\" in Linux."
         },
+        "containerSettings": {
+          "$ref": "#/definitions/TaskContainerSettings",
+          "title": "The settings for the container under which the Job Preparation task runs.",
+          "description": "When this is specified, all directories recursively below the AZ_BATCH_NODE_ROOT_DIR (the root of batch directories on the node) are mapped into the container, all task enviornment variables are mapped into the container, and the task command line is executed in the container."
```

Comment:
This doesn't have the "if container pool then this is required" thing.  Is it optional even on container pools?  If so should we call that out and explain what happens if it's omitted?

Reply:
Also perhaps provide caveats around if you run the JP task in a container then what you need to do to have the effects manifest in other tasks' containers.

Reply:
Yeah, this is intentional but I will clarify.

Reply:
Actually it's awkward to describe what happens when containerSettings is omitted ON containerSettings, because it's basically "the default Azure Batch behavior" -- and when people are trying to understand what the default behavior is they may not be looking on the containerSettings property (because they don't want to specify it)

We could include something like: "If omitted, the task runs as a normal Batch task" - but I'm not sure that's really that helpful...?